cfme/tests/intelligence/chargeback/test_chargeback_long_term_rates.py

```diff
@@ -0,0 +1,473 @@
+def chargeback_report_custom(vm_ownership, assign_custom_rate, interval):
+    # Create a Chargeback report based on a custom rate; Queue the report
+    owner = vm_ownership
+    data = {
+        'menu_name': interval,
+        'title': interval,
+        'base_report_on': 'Chargeback for Vms',
+        'report_fields': ['Memory Used', 'Memory Used Cost', 'Owner',
+        'CPU Used', 'CPU Used Cost',
+        'Disk I/O Used', 'Disk I/O Used Cost',
+        'Network I/O Used', 'Network I/O Used Cost',
+        'Storage Used', 'Storage Used Cost'],
+        'filter': {
+            'filter_show_costs': 'Owner',
+            'filter_owner': owner,
+            'interval_end': 'Today (partial)'
+        }
+    }
+    report = CustomReport(is_candu=True, **data)
+    report.create()
+
+    logger.info('Queuing chargeback report for {} rate'.format(interval))
+    report.queue(wait_for_finish=True)
+
+    yield list(report.get_saved_reports()[0].data.rows)
+    report.delete()
+
+
+@pytest.yield_fixture(scope="module")
+def new_compute_rate(interval):
+    # Create a new Compute Chargeback rate
+    try:
+        desc = 'custom_' + interval
+        compute = rates.ComputeRate(description=desc,
+                    fields={'Used CPU':
+                            {'per_time': interval, 'variable_rate': '720'},
+                            'Used Disk I/O':
+                            {'per_time': interval, 'variable_rate': '720'},
+                            'Used Network I/O':
+                            {'per_time': interval, 'variable_rate': '720'},
+                            'Used Memory':
+                            {'per_time': interval, 'variable_rate': '720'}})
+        compute.create()
+        storage = rates.StorageRate(description=desc,
+                    fields={'Used Disk Storage':
+                            {'per_time': interval, 'variable_rate': '720'}})
+        storage.create()
+        yield desc
+    finally:
+        compute.delete()
+        storage.delete()
+
+
+# Tests to validate costs reported in the Chargeback report for various metrics.
+# The costs reported in the Chargeback report should be approximately equal to the
+# costs estimated in the chargeback_costs_default/chargeback_costs_custom fixtures.
+def test_validate_cpu_usage_cost(chargeback_costs_custom, chargeback_report_custom,
+        interval, provider):
+    """Test to validate CPU usage cost.Calculation is based on custom Chargeback rate.
+
+    """
+    for groups in chargeback_report_custom:
+        if groups["CPU Used Cost"]:
+            estimated_cpu_usage_cost = chargeback_costs_custom['cpu_used_cost']
+            cost_from_report = groups["CPU Used Cost"]
+            cost = re.sub(r'[$,]', r'', cost_from_report)
+            assert estimated_cpu_usage_cost - DEVIATION <= float(cost) \
+                <= estimated_cpu_usage_cost + DEVIATION, \
+                'Estimated cost and report cost do not match'
+            break
```

Comment:
**REQUIRED**
Here and further please use `soft_assert` fixture instead `assert` and `break`.